csrc/fusion_segmenter.cpp

```diff
@@ -141,6 +141,25 @@ void SegmentedGroup::deserialize(
   is_fusion_input_ = buffer->is_fusion_input();
 }
 
+void SegmentedGroup::makeClonedFusion() {
+  auto&& [ir_cloner, fusion_segment] = segmented_fusion_->makeFusion(this);
+  NVF_ERROR(fusion_segment != nullptr, "Failed to create segmented fusion.");
+
+  cloned_fusion_ = std::move(fusion_segment);
+
+  // Map inputs for original fusion to the segmented fusion through IrCloner
+  const std::vector<Val*>& complete_inputs =
+      segmented_fusion_->completeFusion()->inputs();
+  original_inputs_in_cloned_fusion_.reserve(complete_inputs.size());
+  std::transform(
+      complete_inputs.begin(),
+      complete_inputs.end(),
+      std::back_inserter(original_inputs_in_cloned_fusion_),
+      [&complete_to_segment_map = ir_cloner](Val* v) {
+        return complete_to_segment_map.clone(v);
```

Comment:
IIUC, `fusion_segment` returned from this function isn't really changed by this clone. since we should have already cloned it in our segment?

Does it mean we can/should be using `group->getFusion()` instead of `SegmentedFusion::makeFusion(group)` across the board?

Reply:
> Does it mean we can/should be using group->getFusion() instead of SegmentedFusion::makeFusion(group) across the board?

Only if the downstream function does not modify the segmented fusion. It can be a replacement for print functions but not if you need to modify the function.

Reply:
What specifically modifies the fusion and has to be done on the copy? AFAIU, the end state is going to be: after segmentation, 

```
FKR::ctor:
  for each segment:
    create a subfusion for that segment
    heuristics = make initial heuristics for that subfusion

FKR::compileFusionParallel
  for each segment: 
    schedule the associated subfusion according to the initial heuristics
    compile the associated subfusion
```

It'll be unfortunate if we can't reuse the same subfusion for scheduling and compiling and have to copy the fusion.